dev/airflow-github

```diff
@@ -0,0 +1,218 @@
+
+
+def get_commits_between_tags(repo, earlier_tag, later_tag):
+    log_args = ['--format="%H"', earlier_tag + ".." + later_tag]
+    log = repo.git.log(*log_args)
+    return list(log.strip('"').split('"\n"'))
+
+
+def get_issue_status(issue):
+    status = issue.state.capitalize()
+    if status == 'Closed':
+        return 'Resolved'
+    return status
+
+
+def style_issue_status(status):
+    if status in STATUS_COLOR_MAP:
+        return click.style(status[:10].ljust(10), STATUS_COLOR_MAP[status])
+    return status[:10].ljust(10)
+
+
+def get_issue_type(issue):
+    label_prefix = "type:"
+    issue_type = DEFAULT_SECTION_NAME
+    if issue.labels:
+        for label in issue.labels:
+            if label.name.startswith(label_prefix):
+                return label.name.replace(label_prefix, "").strip()
+    return issue_type
+
+
+def print_changelog(sections):
+    for section, lines in sections.items():
+        print(section)
+        print('"' * len(section))
+        for line in lines:
+            print('-', line)
+        print()
+
+
+@click.group()
+def cli():
+    r"""
+    This tool should be used by Airflow Release Manager to verify what Github issue's
+     were merged in the current working branch.
+
+        airflow-github compare <target_version> <github_token>
+    """
+
+
+@cli.command(short_help='Compare a Github target version against git merges')
+@click.argument('target_version')
+@click.argument('github-token', envvar='GITHUB_TOKEN')
+@click.option('--previous-version',
+              'previous_version',
+              help="Specify the previous tag on the working branch to limit"
+                   " searching for few commits to find the cherry-picked commits")
+@click.option('--unmerged', 'show_uncherrypicked_only', help="Show unmerged issues only", is_flag=True)
+def compare(target_version, github_token, previous_version=None, show_uncherrypicked_only=False):
+    repo = git.Repo(".", search_parent_directories=True)
+    commits_to_current_branch = get_commits_to_current_branch(
+        repo, previous_version)
+
+    github_handler = Github(github_token)
+    milestone_issues = list(github_handler.search_issues(
+        f"repo:apache/airflow milestone:\"Airflow {target_version}\""))
+
+    num_cherrypicked = 0
+    num_uncherrypicked = Counter()
+
+    # :<18 says left align, pad to 18
+    # :<50.50 truncates after 50 chars
+    # !s forces as string
+    formatstr = "{id:<8}|{typ!s:<15}|{status!s}|{description:<83.83}|{merged:<6}|{commit:>9.7}"
+
+    print(formatstr.format(
+        id="ISSUE",
+        typ="TYPE",
+        status="STATUS".ljust(10),
+        description="DESCRIPTION",
+        merged="MERGED",
+        commit="COMMIT"))
+
+    for issue in milestone_issues:
+        commit = None
+        for event in issue.get_events():
+            # Checks if event is a commit getting merged into master.
+            if event.event == 'merged':
+                commit = event.commit_id
```

Comment:
This loop is time-consuming, takes a good amount of time on my machine to load all commits.

Reply:
I will push some improvements

Reply:
Fixed in https://github.com/apache/airflow/pull/9270/commits/f93f0ee9832614921436e669141fc779ad5a18e6